test/xds/xds_client_outlier_detection_test.go

```diff
@@ -168,34 +209,35 @@ func (s) TestOutlierDetectionWithOutlier(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	cc, err := grpc.Dial(fmt.Sprintf("xds:///%s", serviceName), grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(resolver))
+	cc, err := grpc.Dial(fmt.Sprintf("xds:///%s", serviceName), grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
 	if err != nil {
 		t.Fatalf("failed to dial local test server: %v", err)
 	}
 	defer cc.Close()
 
 	client := testgrpc.NewTestServiceClient(cc)
-	for i := 0; i < 2000; i++ {
-		// Can either error or not depending on the backend called.
-		if _, err := client.EmptyCall(ctx, &testpb.Empty{}, grpc.WaitForReady(true)); err != nil && !strings.Contains(err.Error(), "some error") {
-			t.Fatalf("rpc EmptyCall() failed: %v", err)
-		}
-		time.Sleep(time.Millisecond)
-	}
 
-	// Backend 1 should've gotten more than 1/3rd of the load as backend 3
-	// should get ejected, leaving only 1 and 2.
-	if count1 < 700 {
-		t.Fatalf("backend 1 should've gotten more than 1/3rd of the load")
+	fullAddresses := []resolver.Address{
+		{Addr: backend1.Address},
+		{Addr: backend2.Address},
+		{Addr: backend3.Address},
 	}
-	// Backend 2 should've gotten more than 1/3rd of the load as backend 3
-	// should get ejected, leaving only 1 and 2.
-	if count2 < 700 {
-		t.Fatalf("backend 2 should've gotten more than 1/3rd of the load")
+	// At first, due to no statistics on each of the backends, the 3
+	// upstreams should all be round robined across.
+	if err = checkRoundRobinRPCs(ctx, client, fullAddresses); err != nil {
+		t.Fatalf("error in expected round robin: %v", err)
+	}
```

Comment:
How important is it to check that RPCs are routed initially to all three backends? I'm guessing it is important. But can we have a separate helper for checking this, which can be much simpler and if we do that, we can use the helper from the `roundrobin` package for the one in line 240.

Reply:
The reason for the initial check is to trigger the ejection (the balancer can count successes/failures), but I see what you're saying. So you want me to get rid of this helper function, and then write a simpler one for the first call, and the second call use that helper (since those backends don't error).

Reply:
Yes.